docker/docker_image_dest.go

```diff
@@ -118,16 +119,16 @@ func (d *dockerImageDestination) PutBlob(digest string, stream io.Reader) error
 	uploadLocation.RawQuery = locationQuery.Encode()
```

Comment:
The `locationQuery.Set("digest", digest)` just above needs to support the case when the digest is not passed by the caller.